### ISSUE TITLE
Add a way to reset settings from scripts

### ISSUE THREAD

**Issue by ArEyeses79** [openplanet-nl/openplanet-issues#230]
Add a way to reset a plugin's settings (without assigning a copy of the default value to the variable).
- A method to reset all settings for the plugin
- A method to reset a category
- A method to reset a single setting

**Comment by ezio416** [comment]
We can reset settings now (not sure when it was added), but to do a specific category we'd need to get all settings and iterate through them to find ones in the category we want. Maybe this can still be added?

**Comment by codecat** [comment]
For completeness, the function in question: https://openplanet.dev/docs/api/Meta/PluginSetting/Reset

Perhaps something could be added, I'm just not convinced it will add much value if we already have this function. What do you think?

**Comment by ezio416** [comment]
I see value in it when we want to create our own settings layout like so - https://github.com/ezio416/tm-warrior-medals/blob/096f5e943e7f060d446830c72b94bd743a2a1f97/src/Settings.as#L51
If I want to add a setting to this category, I have to remember to add another line just to reset it, whereas if I could do `plugin.GetSettingCategory("General").Reset()` I can make the code a lot shorter and not forget things, though this specific syntax would require a new `Meta::SettingCategory` type. `GetSettingCategory` could return an array of settings also but I would prefer the former if this was implemented

**Comment by ArEyeses79** [comment]
ezio can't your use case be done by iterating all settings, checking their category, and if it matches then resetting them? This way you wouldn't need to duplicate the actual setting names.
So its only a qol feature to be able to do this instead of iterating all settings

**Comment by ezio416** [comment]
Yes I can but it's less efficient (not that it really matters in this case)
